NMS-14766: Docs: add more info about provisiond threads (#6850)
OpenNMS/opennms: docs/modules/reference/pages/daemons/daemon-config-files/provisiond.adoc

This pool determines the number of requisitions that can be imported at one time.
Increase this number if you have a large number of requisitions.

> Import threads will check the requisition against the database to determine what add/change/delete operations are necessary.

Reviewer: The Import threads will compare the Nodes in the requisition with the Nodes in the database determine what Node Insert, Update, or Delete operations are necessary. The delete operations are scheduled first, followed by the updates, and finally the inserts. 

(Using the same terminology as in the code. Not wedded to that thinking, though)

Reviewer: Import threads are for the requisitions. The synch button in the UI or the import schedule in provision-configuration.xml

> Import threads will check the requisition against the database to determine what add/change/delete operations are necessary.

Reviewer 2: Just the one update. I'm not clear on why you didn't use the updated content from one of the comments.

> They will also attempt to pull SNMP system information from the node into the database.

Reviewer: ... pull Node attributes from the SNMP system table. (I know there was a reason we did this on the import thread but I can't remember why for the life of me. I know it had something to do with the system object ID but I just can't recall why it had to happen on the import thread. Something about how it gets scheduled for rescans I'm sure.

Author: Maybe so that data is available before it gets to checking policies?

> Once this inventory update is complete, the import thread will schedule a node scan as necessary.

Author: ```suggestion
Once this inventory update is complete, the import thread will schedule the node for scanning or rescanning as necessary.
```

> Once this inventory update is complete, the import thread will schedule the node for scanning/rescanning as necessary.

Reviewer: ... will schedule the Node scan.

Author: "as necessary" based on if the sync is set as all/new/dbOnly
```suggestion
Once this inventory update is complete, the import thread will schedule a node scan as necessary.
```

> Once this inventory update is complete, the import thread will schedule the node for scanning/rescanning as necessary.

Reviewer 2: ```suggestion
Once this inventory update is complete, the import thread will schedule the node for scanning or rescanning as necessary.
```

| Scan
| 10

> | Number of threads to allocate for manual scan/sync requests.

Reviewer: Syncronization is technically on the import thread... this is the insert/update/delete operation phase.

Author: ```suggestion
| Number of threads to allocate for node scan tasks.
```

Reviewer: Scan threads are for the detectors.

> This pool is used for any manual requisition sync tasks and when requesting a rescan of a single node.

Author: ```suggestion
This pool is used for any manual requisition sync tasks and when requesting a rescan of a single node.
Scan threads are in charge of running detectors against a node.
```

Increase this number if you have a large number of nodes in a single requisition.

> Scan threads are in charge of running detectors against a node.

Author: ```suggestion
```

| Rescan
| 10
| Number of threads to allocate for automated rescanning existing nodes.
This pool is used for rescanning nodes in a requisition based on the foreign source definition or the external requisition schedule.

> Rescan threads will spawn a scan thread to process the applicable detectors, so this value typically should be the same as the Scan thread pool size.

Reviewer: Rescan threads are for the Nodes.
